hal/nrf_radio.h

```diff
@@ -0,0 +1,1407 @@
+    NRF_RADIO->BASE1 = address;
+}
+
+__STATIC_INLINE uint32_t nrf_radio_base1_get(void)
+{
+    return NRF_RADIO->BASE1;
+}
+
+__STATIC_INLINE void nrf_radio_prefix0_set(uint32_t prefix0_value)
+{
+    NRF_RADIO->PREFIX0 = prefix0_value;
+}
+
+__STATIC_INLINE uint32_t nrf_radio_prefix0_get(void)
+{
+    return NRF_RADIO->PREFIX0;
+}
+
+__STATIC_INLINE void nrf_radio_prefix1_set(uint32_t prefix1_value)
+{
+    NRF_RADIO->PREFIX1 = prefix1_value;
+}
+
+__STATIC_INLINE uint32_t nrf_radio_prefix1_get(void)
+{
+    return NRF_RADIO->PREFIX1;
+}
+
+__STATIC_INLINE void nrf_radio_txaddress_set(uint8_t txaddress)
+{
+    NRF_RADIO->TXADDRESS = ((uint32_t)txaddress) << RADIO_TXADDRESS_TXADDRESS_Pos;
+}
+
+__STATIC_INLINE uint8_t nrf_radio_txaddress_get(void)
+{
+    return (uint8_t)((NRF_RADIO->TXADDRESS & RADIO_TXADDRESS_TXADDRESS_Msk) >>
+                     RADIO_TXADDRESS_TXADDRESS_Pos);
+}
+
+__STATIC_INLINE void nrf_radio_rxaddresses_set(uint8_t rxaddresses)
+{
+    NRF_RADIO->RXADDRESSES = (uint32_t)(rxaddresses);
+}
+
+__STATIC_INLINE uint8_t nrf_radio_rxaddresses_get(void)
+{
+    return (uint8_t)(NRF_RADIO->RXADDRESSES);
+}
+
+__STATIC_INLINE void nrf_radio_crc_configure(uint8_t                       crc_length,
+                                             nrf_radio_crc_includes_addr_t crc_skip_address,
+                                             uint32_t                      crc_polynominal)
+{
+    NRF_RADIO->CRCCNF = ((uint32_t)crc_length       << RADIO_CRCCNF_LEN_Pos) |
+                        ((uint32_t)crc_skip_address << RADIO_CRCCNF_SKIPADDR_Pos);
+    NRF_RADIO->CRCPOLY = (crc_polynominal << RADIO_CRCPOLY_CRCPOLY_Pos);
+}
+
+__STATIC_INLINE void nrf_radio_crcinit_set(uint32_t crc_init_value)
+{
+    NRF_RADIO->CRCINIT = crc_init_value;
+}
+
+__STATIC_INLINE uint32_t nrf_radio_crcinit_get(void)
+{
+    return NRF_RADIO->CRCINIT;
+}
+
+__STATIC_INLINE void nrf_radio_ifs_set(uint32_t radio_ifs)
+{
+    NRF_RADIO->TIFS = radio_ifs;
+}
+
+__STATIC_INLINE uint32_t nrf_radio_ifs_get(void)
+{
+    return NRF_RADIO->TIFS;
+}
+
+__STATIC_INLINE uint8_t nrf_radio_rssi_sample_get(void)
+{
+    return (uint8_t)(((NRF_RADIO->RSSISAMPLE) & RADIO_RSSISAMPLE_RSSISAMPLE_Msk) >>
+        RADIO_RSSISAMPLE_RSSISAMPLE_Pos);
+}
+
+__STATIC_INLINE nrf_radio_state_t nrf_radio_state_get(void)
+{
+    return (nrf_radio_state_t) NRF_RADIO->STATE;
+}
+
+__STATIC_INLINE void nrf_radio_datawhiteiv_set(uint32_t datawhiteiv)
+{
+    NRF_RADIO->DATAWHITEIV = (datawhiteiv & RADIO_DATAWHITEIV_DATAWHITEIV_Msk);
+}
+
+__STATIC_INLINE uint8_t nrf_radio_datawhiteiv_get(void)
+{
+    return (uint8_t)(NRF_RADIO->DATAWHITEIV & RADIO_DATAWHITEIV_DATAWHITEIV_Msk);
+}
+
+__STATIC_INLINE void nrf_radio_bcc_set(uint32_t radio_bcc)
+{
+    NRF_RADIO->BCC = radio_bcc;
+}
+
+__STATIC_INLINE uint32_t nrf_radio_bcc_get(void)
+{
+    return NRF_RADIO->BCC;
+}
+
+__STATIC_INLINE void nrf_radio_dab_set(uint32_t dab_value, uint8_t segment)
+{
+    NRF_RADIO->DAB[segment] = dab_value;
```

Comment:
`NRFX_ASSERT(segment < 8);`

And add similar check in the consecutive three functions.